libgearman-server/plugins/queue/redis/queue.cc

```diff
@@ -116,7 +116,7 @@ Hiredis::~Hiredis()
 gearmand_error_t Hiredis::initialize()
 {
   int service_port= atoi(service.c_str());
-  if ((_redis= redisConnect("127.0.0.1", service_port)) == NULL)
```

Comment:
There is un-merged Mongo patch which is of a similar flavor (though it came with more tests (eventually)).

Keep in mind, the durable queue is dog slow. There was a gen2 version, which I am unsure of where the code is, that was based off the Baidu extensions to functions. It required workers which acted as agents for the queue.